src/pages/BooksDetail.jsx

```diff
@@ -69,6 +80,38 @@ function BooksDetail({ match }) {
     history.push("/books");
     dispatch(changeDropdown(genre));
   }
+  console.log(favBooks.map((book) => book.id).includes(Number(paramID)));
+
+  const handleFav = async (id) => {
+    console.log("🚀 ~ file: BooksDetail.jsx ~ line 77 ~ handleFav ~ id", id);
+    if (!favBooks.map((book) => book.id).includes(Number(id))) {
+      const docRef = doc(db, "books", `${id}`);
+      const docSnap = await getDoc(docRef);
+      if (docSnap.exists()) {
+        const favBookRef = doc(db, "users", userUID);
+        await updateDoc(favBookRef, {
+          favorites: arrayUnion(docSnap.data()),
+        });
+        const favBooksRef = doc(db, "users", userUID);
+        const favDocSnap = await getDoc(favBooksRef);
+        if (favDocSnap.exists()) {
+          dispatch(setFavorites(favDocSnap.data().favorites));
+        }
+      } else {
+        console.log("No such a doc");
```

Comment:
this

Reply:
not really, I fixed the deployment. so no need to remove them